functions/src/firebase/env.ts

```diff
@@ -30,3 +33,42 @@ export const getTwilioEnv = (): TwilioEnv => {
         apiKeySecret: twilio.apikeysecret,
     }
 }
+
+export const getAppEnv = (): AppEnv => {
+    const { app } = functions.config()
+    if (!app.domain || !app.emailfrom) {
+        throw new InternalServerError('Missing app domain or emailfrom env')
+    }
+    return {
+        domain: app.domain,
+        emailFrom: app.emailfrom,
+    }
+}
```

Comment:
pls consider excusing me for my love for destructuring :)

```js
export const getAppEnv = (): AppEnv => {
     const { app } = functions.config()
     const {domain, emailfrom} = app
     if (!domain || !emailfrom) {
         throw new InternalServerError('Missing app domain or emailfrom env')
     }
     return {
         domain
         emailFrom
     }
 }
```

Reply:
Destructuring has a cost though: https://codeburst.io/es6s-function-destructuring-assignment-is-not-free-lunch-19caacc18137